libsql/src/hrana/mod.rs

```diff
@@ -51,18 +53,50 @@ pub enum HranaError {
     Api(String),
 }
 
-pub struct Statement<T> {
-    client: HttpConnection<T>,
+enum StatementExecutor<T: for<'a> HttpSend<'a>> {
```

Comment:
Its possible to remove the T: HttpSend bound here as well and it should clean up some of the bounds.